sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -359,10 +359,27 @@ case class LambdaVariable(value: String, isNull: String, dataType: DataType) ext
 case class MapObjects(
     function: AttributeReference => Expression,
     inputData: Expression,
-    elementType: DataType) extends Expression {
+    elementType: DataType)(
+    completeFunctionCopy: Option[Expression] = None) extends Expression {
```

Comment:
Why is this a problem?  Is it just `checkAnalysis` is over eager?  Can you just just override `missingInput` or whatever?